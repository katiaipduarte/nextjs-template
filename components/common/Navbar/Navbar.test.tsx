import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import Navbar from './Navbar';

describe('<Navbar /> component Unit Test', () => {
  it('should render 1 <Navbar /> component', () => {
    const { container } = render(
      <Navbar
        mountedComponent={true}
        theme="light"
        themeToggler={jest.fn().mockName('mockedFunction')}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should add active to className in ul, when clicking the faBars icon', () => {
    render(
      <Navbar
        mountedComponent={true}
        theme="light"
        themeToggler={jest.fn().mockName('mockedFunction')}
      />
    );

    const menuBtn = screen.getByTestId('menu-btn');
    const menu = screen.getByTestId('menu');

    expect(menuBtn).toBeTruthy();
    expect(menu.className).toEqual('menu');

    fireEvent.click(screen.getByTestId('mobile-menu-btn'));
    expect(menu.className).toEqual('menu active');
  });
});
