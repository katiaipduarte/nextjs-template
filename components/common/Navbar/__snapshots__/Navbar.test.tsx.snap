// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> component Unit Test should render 1 <Navbar /> component 1`] = `
<div>
  <nav
    class="sc-gsnTZi iBdGQr"
  >
    <ul
      class="menu"
      data-testid="menu"
    >
      <li
        class="logo"
      >
        <a
          href="/"
        >
          App
        </a>
      </li>
      <li
        class="item"
      >
        <a
          href="/about"
        >
          About
        </a>
      </li>
      <li
        class="theme-toggle"
      >
        <button
          aria-label="Toggle dark mode"
          class="sc-bczRLJ cdwfUi"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-moon "
            data-icon="moon"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M32 256c0-123.8 100.3-224 223.8-224c11.36 0 29.7 1.668 40.9 3.746c9.616 1.777 11.75 14.63 3.279 19.44C245 86.5 211.2 144.6 211.2 207.8c0 109.7 99.71 193 208.3 172.3c9.561-1.805 16.28 9.324 10.11 16.95C387.9 448.6 324.8 480 255.8 480C132.1 480 32 379.6 32 256z"
              fill="currentColor"
            />
          </svg>
        </button>
      </li>
      <li
        class="toggle"
        data-testid="menu-btn"
      >
        <button
          aria-label="Toggle dark mode"
          class="sc-bczRLJ cdwfUi"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-moon "
            data-icon="moon"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M32 256c0-123.8 100.3-224 223.8-224c11.36 0 29.7 1.668 40.9 3.746c9.616 1.777 11.75 14.63 3.279 19.44C245 86.5 211.2 144.6 211.2 207.8c0 109.7 99.71 193 208.3 172.3c9.561-1.805 16.28 9.324 10.11 16.95C387.9 448.6 324.8 480 255.8 480C132.1 480 32 379.6 32 256z"
              fill="currentColor"
            />
          </svg>
        </button>
        <a
          aria-label="Click to open mobile menu"
          data-testid="mobile-menu-btn"
          role="button"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-bars "
            data-icon="bars"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 96C0 78.33 14.33 64 32 64H416C433.7 64 448 78.33 448 96C448 113.7 433.7 128 416 128H32C14.33 128 0 113.7 0 96zM0 256C0 238.3 14.33 224 32 224H416C433.7 224 448 238.3 448 256C448 273.7 433.7 288 416 288H32C14.33 288 0 273.7 0 256zM416 448H32C14.33 448 0 433.7 0 416C0 398.3 14.33 384 32 384H416C433.7 384 448 398.3 448 416C448 433.7 433.7 448 416 448z"
              fill="currentColor"
            />
          </svg>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
